### ISSUE TITLE
Cannot find module '@napi-rs/canvas-linux-x64-musl' with node 20

### ISSUE THREAD

**Issue by Khaaz** [Brooooooklyn/canvas#748]
To sum up:

- Node 18
  - windows: OK
  - Docker `node:18-alpine` : OK
- Node 20
  - windows: OK
  - Docker `node:20-alpine` : NOT OK
  
I'm working locally on windows, and it works correctly on node 18 and on node 20.

I just upgraded all my app to node 20. There are no issue on node 20 locally on windows.  
However when I start my app in docker I get the following error:

```
Error: Cannot find module '@napi-rs/canvas-linux-x64-musl'
Require stack:
- /app/node_modules/@napi-rs/canvas/js-binding.js
- /app/node_modules/@napi-rs/canvas/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/app/node_modules/@napi-rs/canvas/js-binding.js:151:31)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/app/node_modules/@napi-rs/canvas/js-binding.js',
    '/app/node_modules/@napi-rs/canvas/index.js'
  ]
}

Node.js v20.9.0
```

This only happens in the container on node 20 as it works perfectly fin locally with node 20 and 18 and in docker with node 18.

The line `"@napi-rs/canvas-linux-x64-musl@0.1.44":` exists in yarn.lock

**Comment by Brooooooklyn** [comment]
@Khaaz you should run yarn install again in Docker

**Comment by Khaaz** [comment]
@Brooooooklyn So I tried again, installed last version of napi and everything.

Apparently, the issue only exists on `node:20-alpine`. I tried it on `node:20` and it works just fine.

Here my complete docker file:

```
# Stage 1: Temporary image to install dependencies
FROM registry.gitlab.com/owlebot/docker-registry/base AS dependencies

WORKDIR /app

COPY package.json ./
COPY yarn.lock ./

RUN yarn install --prod --network-concurrency 1

# Stage 2: Final image with code and installed dependencies
FROM node:20-alpine AS final

WORKDIR /app

# Copy necessary files from the dependencies stage
COPY --from=dependencies /app/node_modules ./node_modules
COPY --from=dependencies /app/package.json ./package.json
COPY --from=dependencies /app/yarn.lock ./yarn.lock

COPY ./src ./src

EXPOSE 3000
CMD [ "yarn", "start:prod" ]
```

`base` Dockerfile looks like this

```
# Stage 0: GIT + SSH auth image
FROM node:20-bookworm AS base

# ARG SSH_GITHUB_KEY

# Install Git and necessary utilities
RUN apt-get update && \
    apt-get install -y git ssh && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

# Copy your SSH key to the container
# RUN echo $SSH_GITHUB_KEY > /root/.ssh/id_ed25519
COPY ./id_ed25519 /root/.ssh/id_ed25519
RUN chmod 600 /root/.ssh/id_ed25519

RUN ssh-keyscan "github.com" >> /root/.ssh/known_hosts
```

TLDR: the issue is not because of the node version but because of the `alpine`.
To sum up:
- it works on node:18-alpine
- it works on node:20
- it doesn't work on node:20-alpine

I can't reopen the issue but I can open another one if you prefer. Looking forward to a quick answer, thanks!!

**Comment by Khaaz** [comment]
@Brooooooklyn to add even more context, it seems that adding 
```
RUN yarn add @napi-rs/canvas-linux-x64-musl
```

in my docker image correctly installs the missing dependency and makes it work.

So i don't know what is exactly the issue but my guess would be that `@napi-rs/canvas-linux-x64-musl` is not installed because already present in the previous step (in the node:20-bookworm image). Which means it doesn't copied to the alpine image.
It worked with node-18 so not sure why it does that

**Comment by Khaaz** [comment]
Ok I enabled more logging when building the image.
I found that

![image](https://github.com/Brooooooklyn/canvas/assets/32022851/634b0f10-76d0-4787-babd-9741b9aa4b30)

So I think the issue is actually a thing when installing the module on node:20.
My best guess is that in node:20 (no alpine), it already has what it needs somehow so it doesn't matter if it can't install the dependency?
But on alpine that's not enough. And since apparently installing it manually (forcing install) works, idk where the issue could be specifically
